core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -107,10 +106,24 @@ object Partition extends KafkaMetricsGroup {
   def apply(topicPartition: TopicPartition,
             time: Time,
             replicaManager: ReplicaManager): Partition = {
+
+    val isrChangeMetrics = new IsrChangeMetrics {
+      override def markExpand(): Unit = {
+        replicaManager.recordIsrChange(topicPartition)
```

Comment:
Hmm.. I had forgotten about this. The purpose of this call is so that we can propagate the ISR change to the controller, but we shouldn't need to do this down the AlterIsr path. This also makes your initial name `IsrChangeListener` a bit more accurate 😅 .